pinot-query-planner/src/main/java/org/apache/pinot/query/planner/DispatchablePlanFragment.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.planner;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.routing.TimeBoundaryInfo;
+import org.apache.pinot.query.routing.PlanFragmentMetadata;
+import org.apache.pinot.query.routing.QueryServerInstance;
+import org.apache.pinot.query.routing.WorkerMetadata;
+
+
+public class DispatchablePlanFragment {
+
+  public static final String TABLE_NAME_KEY = "tableName";
+  public static final String TIME_BOUNDARY_COLUMN_KEY = "timeBoundaryInfo.timeColumn";
+  public static final String TIME_BOUNDARY_VALUE_KEY = "timeBoundaryInfo.timeValue";
+  private final PlanFragment _planFragment;
+  private final List<WorkerMetadata> _workerMetadataList;
+
+  // This is used at broker stage - we don't need to ship it to the server.
+  private Map<QueryServerInstance, List<Integer>> _serverInstanceToWorkerIdMap;
+
+  // used for table scan stage - we use ServerInstance instead of VirtualServer
+  // here because all virtual servers that share a server instance will have the
+  // same segments on them
+  private Map<Integer, Map<String, List<String>>> _workerIdToSegmentsMap;
+  private Map<String, String> _customProperties;
```

Comment:
can we add a comment indicating that this is the generic property format that will be used to pass info to stageMetadat on server?